src/core/connection.ts

```diff
@@ -156,13 +156,30 @@ export class DefaultConnection extends EventEmitter<ConnectionEventsMap> impleme
       // If we're in the disconnected state, assume it's transient and set a timeout to propagate the change
       if (chatState === ConnectionStatus.Disconnected && !this._transientTimeout) {
         this._transientTimeout = setTimeout(() => {
+          this._logger.debug('transient disconnect timeout reached');
           this._transientTimeout = undefined;
-          this._applyStatusChange(stateChange);
+
+          // When we apply the status change, we should apply whatever the current state is at the time
+          this._applyStatusChange({
+            ...stateChange,
+            current: this._mapAblyStatusToChat(this._connection.state),
+          });
         }, TRANSIENT_TIMEOUT);
         return;
       }
 
-      // If we're in any state other than disconnected, and we have a transient timeout, clear it
+      // If we're in the connecting state, or disconnected state, and we have a transient timeout, we should ignore it -
+      // if we can reach connected in a reasonable time, we can assume the disconnect was transient and suppress the
+      // change
+      if (
```

Comment:
may as well nest if statements here to avoid repeating the `if (this._transientTimeout)` condition:
```ts
if (this._transientTimeout) {
  if (chatState === ConnectionStatus.Connecting || chatState === ConnectionStatus.Disconnected) {
    return;
  }
  
  this.cancelTransientTimeout();
}
```

Reply:
Done